packages/webapp/components/layouts/ProfileLayout/index.tsx

```diff
@@ -96,14 +91,18 @@ export default function ProfileLayout({
       enabled: !!initialProfile,
     },
   );
-
   // Needed because sometimes initialProfile is defined and fetchedProfile is not
   const profile = fetchedProfile ?? initialProfile;
-  const isCurrentUserProfile = profile && profile.id === user?.id;
-
+  useEffect(() => {
+    if (profile) {
+      const purify = DOMPurify(window);
+      setTwitterHandle(sanitizeOrNull(purify, profile.twitter));
+      setGithubHandle(sanitizeOrNull(purify, profile.github));
+      setHashnodeHandle(sanitizeOrNull(purify, profile.hashnode));
+      setPortfolioLink(sanitizeOrNull(purify, profile.portfolio));
```

Comment:
I don't think you need effect + state here, you can just `useMemo`.

Reply:
You mean memo on the profile or the actual states?

Reply:
Hmm probably indeed, it's pretty much a edge case anyway not used that much.
I'll log it as a extra ticket if that's ok.

Reply:
Oh, right. Yep. One less re-render.